src/io/flutter/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -262,17 +263,21 @@ public void scheduleConnect() {
       // because "flutter run" has already connected to it.
       final VmService vmService;
       final VmOpenSourceLocationListener vmOpenSourceLocationListener;
-      try {
-        vmService = VmService.connect(url);
-        vmOpenSourceLocationListener = VmOpenSourceLocationListener.connect(url);
-      }
-      catch (IOException | RuntimeException e) {
-        onConnectFailed("Failed to connect to the VM observatory service at: " + url + "\n"
-                        + e.toString() + "\n" +
-                        formatStackTraces(e));
-        return;
+      // TODO(github.com/flutter/flutter-intellij/issues/3293) The following check disables the WebSocket connection for all FlutterWeb
```

Comment:
please add more granularity to this issue. How does this test fail. 
If it is needed, check based on whether the app is a FlutterWebApp not based on the device being null.